app/enterprise.go

```diff
@@ -35,9 +36,9 @@ func RegisterDataRetentionInterface(f func(*App) einterfaces.DataRetentionInterf
 	dataRetentionInterface = f
 }
 
-var elasticsearchInterface func(*App) einterfaces.ElasticsearchInterface
+var elasticsearchInterface func(*App) searchengine.SearchEngineInterface
 
-func RegisterElasticsearchInterface(f func(*App) einterfaces.ElasticsearchInterface) {
+func RegisterElasticsearchInterface(f func(*App) searchengine.SearchEngineInterface) {
```

Comment:
I think is not right, because this is the mechanism to register ElasticSearch, no any other search engine, this mechanism is to plugin the enterprise code. This `ElasticSearchInterface` specific, not `SearchEngineInterface` in general.